src/cpu/aarch64/jit_uni_reorder.cpp

```diff
@@ -170,13 +170,18 @@ struct jit_uni_reorder_kernel_f32_t : public kernel_t, public jit_generator {
                 || (p.itype == bf16 && p.otype == f32 && mayiuse_bf16()
                         && p.beta == 0.f);
 
+        bool f16_ok = (p.itype != f16 && p.otype != f16)
+                || (p.itype == f16 && p.otype == f32 && p.beta == 0.f);
```

Comment:
You likely want to build your condition using implications:
```
&& IMPLICATION(is_f16, f16_ok)
&& ...
```
Then those `(p.itype != f16 && p.otype != f16)` wouldn't be needed which should give more clarity.